app/src/main/java/nl/tudelft/pixelperfect/client/ClientListener.java

```diff
@@ -22,34 +23,81 @@
-                    break;
-                case "AsteroidFieldEvent":
-                    mission = new AsteroidFieldEvent(eve.getID(), "", "", eve.getTime(), eve.getDuration(), 0);
-                    LocationLabActivity.updateEventLog(mission);
-                    break;
-                case "HostileShipEvent":
-                    mission = new HostileShipEvent(eve.getID(), "", "", eve.getTime(), eve.getDuration(), 0);
-                    LocationArmoryActivity.updateEventLog(mission);
-                    break;
-                case "PlasmaLeakEvent":
-                    mission = new PlasmaLeakEvent(eve.getID(), "", "", eve.getTime(), eve.getDuration(), 0);
-                    LocationEngineroomActivity.updateEventLog(mission);
-                    break;
-                default:
-                    break;
-            }
+            updateEventLog(source, message);
         } else if (message instanceof RoleChosenMessage) {
-            RoleChosenMessage roleMessage = (RoleChosenMessage) message;
-            roleMessage.getRole().updateButtons();
+            updateRoleAvailabilty(message);
+        }
+    }
+
+    /**
+     * Updates the eventlog of the client.
+     *
+     * @param source the source of the message.
+     * @param message the message received.
+     */
+    public void updateEventLog(Client source, Message message) {
+        Event mission;
+        EventsMessage eve = (EventsMessage) message;
+        System.out.println("Client #"+source.getId()+" received event: '"+eve.getType()+"'");
+        switch (eve.getType()) {
+            case "FireEvent":
+                mission = new FireEvent(eve.getID(), "", "", eve.getTime(), eve.getDuration(), 0);
+                LocationArmoryActivity.updateEventLog(mission);
+                break;
+            case "AsteroidFieldEvent":
+                mission = new AsteroidFieldEvent(eve.getID(), "", "", eve.getTime(), eve.getDuration(), 0);
+                LocationLabActivity.updateEventLog(mission);
+                break;
+            case "HostileShipEvent":
+                mission = new HostileShipEvent(eve.getID(), "", "", eve.getTime(), eve.getDuration(), 0);
+                LocationArmoryActivity.updateEventLog(mission);
+                break;
+            case "PlasmaLeakEvent":
+                mission = new PlasmaLeakEvent(eve.getID(), "", "", eve.getTime(), eve.getDuration(), 0);
+                LocationEngineroomActivity.updateEventLog(mission);
+                break;
+            default:
+                break;
+        }
+    }
+
+    /**
+     * Updates the view of the roles in the RoleActivity so that no more than 1 player may choose
+     * the same role.
+     *
+     * @param message the message received.
+     */
+    public void updateRoleAvailabilty(Message message) {
+        RoleChosenMessage roleMessage = (RoleChosenMessage) message;
+        switch (roleMessage.getRole()) {
```

Comment:
Switch statements like these will definitely be disapproved by the TA. I will approve this for now though, since we really need to focus on functionality now, and refactor it later.